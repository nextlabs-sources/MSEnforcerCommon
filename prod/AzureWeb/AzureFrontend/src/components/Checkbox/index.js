import React from 'react'
import propTypes from 'prop-types'

const Checkbox = ({ checked = false, handler, style }) => {
    const innerStyle = {
        display: 'inline-block',
        verticalAlign: 'top',
        border: checked ? '2px solid #06c' : '2px solid #999',
        borderRadius: '3px',
        boxSizing: 'border-box',
        backgroundColor: checked ? '#06c' : '#fff',
        backgroundClip: 'content-box',
        cursor: 'pointer',
    }

    const mergeStyle = {
        ...innerStyle,
        ...style,
    }

    const clickHandler = (e) => {
        e.stopPropagation()

        if (handler) {
            handler(!checked)
        }
    }
    
    return (
        <div style={mergeStyle} onClick={clickHandler} role='checkbox' aria-checked={checked} tabIndex={-1}>
            <svg focusable="false" width="100%" height="100%" viewBox="0 0 24 24" aria-hidden="true" preserveAspectRatio='xMinYMin meet'>
                <path fill='#fff' d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"/>
            </svg>
        </div>
    )
}

export default Checkbox

Checkbox.propTypes = {
    checked: propTypes.bool,
    handler: propTypes.func.isRequired,
    style: propTypes.objectOf(propTypes.string),
}
