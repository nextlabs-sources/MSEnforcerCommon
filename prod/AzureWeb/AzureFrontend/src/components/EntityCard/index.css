.table-uncollapse {
    overflow: hidden;
}
.table-uncollapse-enter {
    max-height: 0;
    opacity: 0.01;
}
.table-uncollapse-enter-active {
    max-height: 360px;
    opacity: 1;
    transition: all .2s ease-in-out;
}
.table-uncollapse-exit {
    max-height: 360px;
    opacity: 1;
}
.table-uncollapse-exit-active {
    max-height: 0;
    opacity: 0.01;
    transition: all .2s ease-in-out;
}