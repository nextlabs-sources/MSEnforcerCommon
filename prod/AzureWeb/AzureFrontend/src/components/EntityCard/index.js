import React, { Component } from 'react'
import propTypes from 'prop-types'
import Checkbox from '../Checkbox'
import './index.css'

class EntityCard extends Component {

    static defaultProps = {
        entity: [],
    }

    static contextTypes = {
        server: propTypes.string,
        database: propTypes.string,
        checkHandler: propTypes.func,
    }

    constructor(props) {
        super(props)
        this.state = {
            collapsed: true,
        }
        this.clickHandler = this.clickHandler.bind(this)
        this.checkHandler = this.checkHandler.bind(this)
    }

    clickHandler(e) {
        this.setState({
            collapsed: !this.state.collapsed,
        })
    }

    checkHandler(checked) {
        const { entity } = this.props
        const { server, database, checkHandler } = this.context

        checkHandler(server, database, entity.schema, entity.name, checked)
    }

    render() {
        const { entity } = this.props

        return (
            <div>
                <div
                    style={{
                        height: '100px',
                        padding: '10px',
                        borderRadius: '4px',
                        boxShadow: '0 0 4px 1px #ccc',
                        margin: '32px 20px',
                        fontSize: '0',
                        cursor: 'pointer',
                        backgroundColor: this.state.collapsed ? '#fff' : 'rgb(250, 250, 250)',
                    }}
                    onClick={this.clickHandler}
                    role='button'
                >
                    <div style={{
                        textAlign: 'center', display: 'inline-block', padding: '1.2em 24px 24px', fontSize: '16px', verticalAlign: 'top',
                    }}
                    >
                        <Checkbox checked={entity.willbeEnforced} handler={this.checkHandler} style={{ width: '24px', height: '24px' }} />
                    </div>
                    <div style={{ textAlign: 'left', display: 'inline-block', fontSize: '16px' }}>
                        <h3 style={{
                            fontWeight: '400', fontSize: 'inherti', color: '#333', marginTop: '1em',
                        }}
                        >
                            {entity.name}
                        </h3>
                        <p style={{ fontSize: '11px', color: '#999' }}>{entity.description}</p>
                    </div>
                </div>
                {/* <CSSTransition classNames='table-uncollapse' in={false} timeout={300} unmountOnExit>
                    <div style={{ margin: '0 20px' }}>
                        <AttributeTable server={server} database={database} table={entity.name} attributes={entity.attributes} />
                    </div>
                </CSSTransition> */}
            </div>
        )
    }
}

export default EntityCard

EntityCard.propTypes = {
    entity: propTypes.objectOf(propTypes.oneOfType([propTypes.any, propTypes.objectOf({
        name: propTypes.string.isRequired,
        description: propTypes.string,
        enforced: propTypes.bool,
        willbeEnforced: propTypes.bool,
    })])).isRequired,
}
