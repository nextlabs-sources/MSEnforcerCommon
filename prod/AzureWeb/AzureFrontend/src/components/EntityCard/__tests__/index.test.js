import React from 'react'
import propTypes from 'prop-types'
import renderer from 'react-test-renderer'
import EntityCard from '../index'
import Checkbox from '../../Checkbox'

const mockContext = {
  server: 'nextlabs.com',
  database: 'cdc',
  checkHandler: jest.fn(),
}
const mockEntity = {
  name: 'Account',
  description: 'Account Mock',
  enforced: true,
  willBeEnforced: true,
}
class ContextProvider extends React.Component {

  static childContextTypes = {
    server: propTypes.string,
    database: propTypes.string,
    checkHandler: propTypes.func,
  }

  getChildContext() {
    const { mockHandler } = this.props
    return {
      ...mockContext,
      checkHandler: mockHandler,
    }
  }

  render() {
    return React.Children.only(this.props.children)
  }
}

describe('EntityCard', () => {
  it('should render checked entity', () => {
    const mockCheckHandler = jest.fn()
    const tree = renderer.create(<ContextProvider mockHandler={mockCheckHandler}><EntityCard entity={{ ...mockEntity }} /></ContextProvider>)
    const instance = tree.root
    const checkboxIns = instance.findByType(Checkbox)

    checkboxIns.props.handler(true)

    expect(tree.toJSON()).toMatchSnapshot()
    expect(checkboxIns).not.toBe(undefined)
    expect(mockCheckHandler.mock.calls.length).toBe(1)
  })

  it('should render unchecked entity', () => {
    const mockCheckHandler = jest.fn()
    const tree = renderer.create(<ContextProvider mockHandler={mockCheckHandler}><EntityCard entity={{ ...mockEntity, enforced: false, willBeEnforced: false }} /></ContextProvider>)
    const instance = tree.root
    const checkboxIns = instance.findByType(Checkbox)

    checkboxIns.props.handler(true)

    expect(tree.toJSON()).toMatchSnapshot()
    expect(checkboxIns).not.toBe(undefined)
    expect(mockCheckHandler.mock.calls.length).toBe(1)
  })
})