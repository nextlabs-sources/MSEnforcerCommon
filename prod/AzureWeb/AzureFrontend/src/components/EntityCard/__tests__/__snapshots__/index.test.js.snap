// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EntityCard should render checked entity 1`] = `
<div>
  <div
    onClick={[Function]}
    role="button"
    style={
      Object {
        "backgroundColor": "#fff",
        "borderRadius": "4px",
        "boxShadow": "0 0 4px 1px #ccc",
        "cursor": "pointer",
        "fontSize": "0",
        "height": "100px",
        "margin": "32px 20px",
        "padding": "10px",
      }
    }
  >
    <div
      style={
        Object {
          "display": "inline-block",
          "fontSize": "16px",
          "padding": "1.2em 24px 24px",
          "textAlign": "center",
          "verticalAlign": "top",
        }
      }
    >
      <div
        aria-checked={false}
        onClick={[Function]}
        role="checkbox"
        style={
          Object {
            "backgroundClip": "content-box",
            "backgroundColor": "#fff",
            "border": "2px solid #999",
            "borderRadius": "3px",
            "boxSizing": "border-box",
            "cursor": "pointer",
            "display": "inline-block",
            "height": "24px",
            "verticalAlign": "top",
            "width": "24px",
          }
        }
        tabIndex={-1}
      >
        <svg
          aria-hidden="true"
          focusable="false"
          height="100%"
          preserveAspectRatio="xMinYMin meet"
          viewBox="0 0 24 24"
          width="100%"
        >
          <path
            d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
            fill="#fff"
          />
        </svg>
      </div>
    </div>
    <div
      style={
        Object {
          "display": "inline-block",
          "fontSize": "16px",
          "textAlign": "left",
        }
      }
    >
      <h3
        style={
          Object {
            "color": "#333",
            "fontSize": "inherti",
            "fontWeight": "400",
            "marginTop": "1em",
          }
        }
      >
        Account
      </h3>
      <p
        style={
          Object {
            "color": "#999",
            "fontSize": "11px",
          }
        }
      >
        Account Mock
      </p>
    </div>
  </div>
</div>
`;

exports[`EntityCard should render unchecked entity 1`] = `
<div>
  <div
    onClick={[Function]}
    role="button"
    style={
      Object {
        "backgroundColor": "#fff",
        "borderRadius": "4px",
        "boxShadow": "0 0 4px 1px #ccc",
        "cursor": "pointer",
        "fontSize": "0",
        "height": "100px",
        "margin": "32px 20px",
        "padding": "10px",
      }
    }
  >
    <div
      style={
        Object {
          "display": "inline-block",
          "fontSize": "16px",
          "padding": "1.2em 24px 24px",
          "textAlign": "center",
          "verticalAlign": "top",
        }
      }
    >
      <div
        aria-checked={false}
        onClick={[Function]}
        role="checkbox"
        style={
          Object {
            "backgroundClip": "content-box",
            "backgroundColor": "#fff",
            "border": "2px solid #999",
            "borderRadius": "3px",
            "boxSizing": "border-box",
            "cursor": "pointer",
            "display": "inline-block",
            "height": "24px",
            "verticalAlign": "top",
            "width": "24px",
          }
        }
        tabIndex={-1}
      >
        <svg
          aria-hidden="true"
          focusable="false"
          height="100%"
          preserveAspectRatio="xMinYMin meet"
          viewBox="0 0 24 24"
          width="100%"
        >
          <path
            d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
            fill="#fff"
          />
        </svg>
      </div>
    </div>
    <div
      style={
        Object {
          "display": "inline-block",
          "fontSize": "16px",
          "textAlign": "left",
        }
      }
    >
      <h3
        style={
          Object {
            "color": "#333",
            "fontSize": "inherti",
            "fontWeight": "400",
            "marginTop": "1em",
          }
        }
      >
        Account
      </h3>
      <p
        style={
          Object {
            "color": "#999",
            "fontSize": "11px",
          }
        }
      >
        Account Mock
      </p>
    </div>
  </div>
</div>
`;
